compositor_common/src/scene.rs

```diff
@@ -22,6 +22,24 @@ pub struct InputId(pub NodeId);
 #[derive(Debug, Clone, Hash, PartialEq, Eq, Serialize, Deserialize)]
 pub struct OutputId(pub NodeId);
 
+impl Display for InputId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0 .0.fmt(f)
+    }
+}
+
+impl Display for OutputId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0 .0.fmt(f)
+    }
+}
+
+impl Display for NodeId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
```

Comment:
Consider removing it, deriving `Debug` and using `{0:?}` syntax in error logs:
```rust 
#[derive(Debug, thiserror::Error)]
pub enum RegisterInputError {
    #[error("Input {0:?} is already registered.")]
    AlreadyRegistered(InputId),
}
```

I'm not saying that this is necessarily bad. If you find this more descriptive you can leave it as it is.

Reply:
Those errors will be user facing, so I want to display "Input input1 is already registered." and not "Input InputId(NodeId(input1)) is already registered."